apis/core/v1beta1/function_webhook.go

```diff
@@ -38,4 +43,418 @@ var _ webhook.Defaulter = &Function{}
 // Default implements webhook.Defaulter so a webhook will be registered for the type
 func (r *Function) Default() {
 	functionlog.Info("default", "name", r.Name)
+	if r.Spec.Version == nil || *r.Spec.Version == "" {
+		version := "latest"
+		r.Spec.Version = &version
+	}
+}
+
+// +kubebuilder:webhook:path=/validate-core-openfunction-io-v1beta1-function,mutating=false,failurePolicy=fail,groups=core.openfunction.io,resources=functions,verbs=create;update,versions=v1beta1,name=vfunctions.of.io,sideEffects=None,admissionReviewVersions=v1
+var _ webhook.Validator = &Function{}
+
+func (r *Function) ValidateCreate() error {
+	functionlog.Info("validate create", "name", r.Name)
+	return r.Validate()
+}
+
+func (r *Function) ValidateUpdate(old runtime.Object) error {
+	functionlog.Info("validate update", "name", r.Name)
+	return r.Validate()
+}
+
+func (r *Function) ValidateDelete() error {
+	functionlog.Info("validate delete", "name", r.Name)
+	return nil
+}
+
+func (r *Function) Validate() error {
+	if r.Spec.Image == "" {
+		return errors.New("spec.image must be set")
```

Comment:
A k8s-ized error is required here, and `filed.Required` is more appropriate.

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go#L184